golang/cosmos/x/swingset/abci.go

```diff
@@ -47,13 +44,11 @@ func BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock, keeper Keeper) erro
 		ChainID:     ctx.ChainID(),
 		Params:      keeper.GetParams(ctx),
 	}
-	b, err := json.Marshal(action)
-	if err != nil {
-		return sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+	_, err := keeper.BlockingSend(ctx, action)
+	if err == nil {
+		err = keeper.PushAction(ctx, action)
```

Comment:
I'm confused that this action is being both send over the bridge and stored in the action queue - will it be double-interpreted on the JS side? Worthy of a comment either way. Also comment on the significance of sending first here, enqueueing first in `EndBlock`, and not enqueueing for `CommitBlock`.

Reply:
You identified what was a wart in the block-manager.js.  I've updated it now.